deepinv/optim/optim_iterators/spectral_methods.py

```diff
@@ -0,0 +1,107 @@
+import torch
+
+from .optim_iterator import OptimIterator, fStep, gStep
+
+
+class SMIteration(OptimIterator):
+    r"""
+    Iterator for Spectral Methods.
+    
+    Class for a single iteration of the Spectral Methods algorithm to find the principal eigenvector of the regularized weighted covariance matrix:
+    
+    .. math::
+        \begin{equation*}
+        M = A^H diag(T(y)) A + \lambda I,
+        \end{equation*}
+    
+    where :math:`A` is the forward matrix of the phase retrieval operator, :math:`T(\cdot)` is a preprocessing function for the measurements, and :math:`I` is the identity matrix of corresponding dimensions. Parameter :math:`\lambda` tunes the strength of regularization.
+
+    The iteration is given by
+
+    .. math::
+        \begin{equation*}
+        \begin{aligned}
+        x_{k+1} &= (M + \lambda I)x_k \\
+        x_{k+1} &= \operatorname{prox}_{\gamma g}(x_{k+1}),
+        \end{aligned}
+        \end{equation*}
+
+
+    where :math:`\gamma` is a stepsize that should satisfy :math:`\lambda \gamma \leq 2/\operatorname{Lip}(\|\nabla f\|)`.
+    """
+
+    def __init__(self, lamb, n_iter=50, preprocessing=lambda x: x, **kwargs):
+        super(SMIteration, self).__init__()
+        self.n_iter = n_iter
+        self.f_step = fStepSM(lamb, preprocessing=preprocessing, **kwargs)
+        self.g_step = gStepSM(**kwargs)
+
+    def forward(self, x, cur_prior, cur_params, y, physics):
+        r"""
+        Single iteration of the spectral method.
+
+        :param dict x: the current iterate :math:`x_k`.
+        :param deepinv.optim.prior cur_prior: Instance of the Prior class defining the current prior.
+        :param dict cur_params: Dictionary containing the current parameters of the algorithm.
+        :param torch.Tensor y: Input data.
+        :param deepinv.physics physics: Instance of the physics containing the forward operator.
+        :return: The new iterate :math:`x_{k+1}`.
+        """
+        x = self.f_step(x, y, physics)
+        x = self.g_step(x, cur_prior, cur_params)
+        return x
+
+
+class fStepSM(fStep):
+    r"""
+    Spectral Methods fStep module.
+    """
+
+    def __init__(
+        self, lamb=10, preprocessing=lambda x: torch.max(1 - 1 / x, -5), **kwargs
```

Comment:
the default value of the preprocessing is different between here and line 33. I would suggest using the optimal one in both (and adding lamb = 10 in line 33 too)